src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/LocalVCPrePushHook.java

```diff
@@ -103,6 +111,32 @@ public void onPreReceive(ReceivePack receivePack, Collection<ReceiveCommand> com
                 }
             }
 
+            // ----------- Check for large blobs (>10MB) -----------
+            ObjectReader reader = repository.newObjectReader();
+            RevWalk revWalk = new RevWalk(reader);
+            RevCommit newCommit = revWalk.parseCommit(command.getNewId());
+
+            // Use TreeWalk to walk all new files in the tree
+            TreeWalk treeWalk = new TreeWalk(reader);
+            treeWalk.addTree(newCommit.getTree());
+            treeWalk.setRecursive(true);
+
+            while (treeWalk.next()) {
+                ObjectId objectId = treeWalk.getObjectId(0);
+                ObjectLoader loader = reader.open(objectId);
+
+                if (loader.getType() == Constants.OBJ_BLOB && loader.getSize() > MAX_BLOB_SIZE_BYTES) {
+                    command.setResult(ReceiveCommand.Result.REJECTED_OTHER_REASON,
+                            String.format("File '%s' exceeds 10MB size limit (%.2f MB)", treeWalk.getPathString(), loader.getSize() / (1024.0 * 1024.0)));
+                    reader.close();
+                    revWalk.close();
+                    return;
```

Comment:
`git.close();` is missing here.
You could also just replace this with a break instead of returning here.
```suggestion
                    break
```